### PR TITLE
Insert stored fee recipient for `ForkchoiceUpdated` call

### PR DESCRIPTION
**Pull request by terencechain**
This PR uses the stored fee recipient in DB to construct `PayloadAttributes` and to make a `ForkchoiceUpdated` call. If the fee recipient is in DB then it uses the default fee recipient

### CODE REVIEW

File: beacon-chain/node/config.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func configureInteropConfig(cliCtx *cli.Context) {
 func configureExecutionSetting(cliCtx *cli.Context) {
 	if cliCtx.IsSet(flags.FeeRecipient.Name) {
 		c := params.BeaconConfig()
-		c.FeeRecipient = common.HexToAddress(cliCtx.String(flags.FeeRecipient.Name))
+		c.DefaultFeeRecipient = common.HexToAddress(cliCtx.String(flags.FeeRecipient.Name))
```
Comment (james-prysm) [inline beacon-chain/node/config.go:111]: I called my flag SuggestedFeeRecipient similar to lighthouse, wondering if i should change it to default as well.

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go [hunk 2]
```diff
@@ -24,7 +25,7 @@ import (
 
 // This returns the execution payload of a given slot. The function has full awareness of pre and post merge.
 // The payload is computed given the respected time of merge.
-func (vs *Server) getExecutionPayload(ctx context.Context, slot types.Slot) (*enginev1.ExecutionPayload, error) {
+func (vs *Server) getExecutionPayload(ctx context.Context, slot types.Slot, vIdx types.ValidatorIndex) (*enginev1.ExecutionPayload, error) {
```
Comment (james-prysm) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:28]: this makes sense now

Comment (james-prysm) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:28]: this makes sense now

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go [hunk 3]
```diff
@@ -94,10 +95,21 @@ func (vs *Server) getExecutionPayload(ctx context.Context, slot types.Slot) (*en
...
 		FinalizedBlockHash: finalizedBlockHash,
 	}
+
+	feeRecipient := params.BeaconConfig().DefaultFeeRecipient
+	recipient, err := vs.BeaconDB.FeeRecipientByValidatorID(ctx, vIdx)
+	switch err == nil {
+	case true:
+		feeRecipient = recipient
+	case errors.As(err, kv.ErrNotFoundFeeRecipient): // If fee recipient is not found, use the default fee recipient.
+		logrus.WithError(err).WithField("validatorIndex", vIdx).Error("Fee recipient not found. Using default fee recipient.")
```
Comment (james-prysm) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:105]: wonder if you should print what the default fee recipient address is.

Comment (james-prysm) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:105]: wonder if you should print what the default fee recipient address is.

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go]: done!

File: beacon-chain/node/config.go (hunk 1 above)
Comment (james-prysm) [inline beacon-chain/node/config.go:111]: I think we should validate here using `^0x[0-9a-fA-F]{40}$` for the hex

File: beacon-chain/node/config.go [hunk 4]
```diff
@@ -105,10 +107,17 @@ func configureInteropConfig(cliCtx *cli.Context) {
...
 }
 
-func configureExecutionSetting(cliCtx *cli.Context) {
-	if cliCtx.IsSet(flags.FeeRecipient.Name) {
-		c := params.BeaconConfig()
-		c.FeeRecipient = common.HexToAddress(cliCtx.String(flags.FeeRecipient.Name))
-		params.OverrideBeaconConfig(c)
+func configureExecutionSetting(cliCtx *cli.Context) error {
+	if !cliCtx.IsSet(flags.FeeRecipient.Name) {
+		return nil
```
Comment (james-prysm) [inline beacon-chain/node/config.go:112]: maybe we can add a log here to inform the user that the validator will be sending the address or else it defaults to burn address.

**Review by james-prysm — APPROVED** [review]
LGTM besides the items discussed in discord for a later PR

**Pull request merged**